vllm/model_executor/layers/fused_moe/fused_moe.py

```diff
@@ -203,14 +203,15 @@ def moe_align_block_size(
     - The padding ensures that the total number of tokens is now divisible
         by block_size for proper block matrix operations.
     """
-    sorted_ids = torch.empty(
-        (topk_ids.numel() + num_experts * (block_size - 1), ),
-        dtype=torch.int32,
-        device=topk_ids.device)
-    expert_ids = torch.empty((topk_ids.numel() + num_experts, ),
+    max_num_tokens_padded = topk_ids.numel() + num_experts * (block_size - 1)
+    sorted_ids = torch.empty((max_num_tokens_padded, ),
                              dtype=torch.int32,
                              device=topk_ids.device)
     sorted_ids.fill_(topk_ids.numel())
+    num_m_blocks = triton.cdiv(max_num_tokens_padded, block_size)
```

Comment:
Maybe call this `max_num_m_blocks` to be consistent with the above? I think the true number of blocks can be smaller than that.

Reply:
Good point. Fixed!